### PR TITLE
Adds 'unsigned' flag to &nrformats.

### PR DESCRIPTION
**Pull request by rbtnn**
I want to add unsigned flag to &nrformats. Because Its the behavior is NOT intuitive when increase/decrease a date(, zipcode, etc..) splited by '-'.

### CODE REVIEW

File: src/testdir/test_increment.vim [hunk 1]
```diff
@@ -796,4 +796,54 @@ func Test_increment_special_char()
   call assert_beeps("normal \<C-X>")
 endfunc
 
+" Try incrementing/decrementing a number when nrformats contains unsigned
+func Test_unsigned()
```
Comment (ichizok) [inline src/testdir/test_increment.vim]: I think should name including `increment` same as other tests; e.g. Test_increment_unsigned or Test_unsinged_increment

File: src/testdir/test_increment.vim [hunk 2]
```diff
@@ -796,4 +796,54 @@ func Test_increment_special_char()
...
+
+  call setline(1, '-1')
+  exec "norm! gg0\<C-A>"
+  call assert_equal('-2', getline(1))
+
+  call setline(1, '-3')
+  exec "norm! gg0\<C-A>"
+  call assert_equal('-4', getline(1))
+
+  set nrformats-=unsigned
```
Comment (ichizok) [inline src/testdir/test_increment.vim:840]: This restoring 'nrformats' is no need because it is set to default in SetUp().

Comment (rbtnn) [reply src/testdir/test_increment.vim:840]: I fixed it. Thank you for reviewing this PR.

File: runtime/doc/options.txt [hunk 3]
```diff
@@ -5427,6 +5427,13 @@ A jump table for the options with a short description can be found at |Q_op|.
 	bin	If included, numbers starting with "0b" or "0B" will be
 		considered to be binary.  Example: Using CTRL-X on
 		"0b1000" subtracts one, resulting in "0b0111".
+	unsigned    If included, numbers is recognized to be unsigned number.
+		Thus the numbers does not contained negative sign('-').
```
Comment (k-takata) [inline runtime/doc/options.txt]:
```suggestion
		Thus the numbers does not contained negative sign ('-').
```

File: src/testdir/test_increment.vim [hunk 4]
```diff
@@ -796,4 +796,52 @@ func Test_increment_special_char()
   call assert_beeps("normal \<C-X>")
 endfunc
 
+" Try incrementing/decrementing a number when nrformats contains unsigned
+func Test_increment_unsigned()
+  set nrformats+=unsigned
+
+  " 18446744073709551615 == pow(2, 64) - 1
+  if 64 == v:numbersize
```
Comment (k-takata) [inline src/testdir/test_increment.vim]: Now `v:numbersize` is always 64.

File: src/testdir/test_increment.vim (hunk 2 above)
Comment (chrisbra) [reply src/testdir/test_increment.vim:840]: FWIW: I prefer explicitly resetting of the `nrformats` just to make sure it is explicit and when a somebody copy pastes this as a new test, it does not depend on the Setup and TearDown function.

File: runtime/doc/options.txt (hunk 3 above)
Comment (chrisbra) [reply runtime/doc/options.txt]: I think better would be something like:

`If included, numbers are recognized as unsigned. Thus a leading negative sign won't be considered as part of the number.`

File: src/testdir/test_increment.vim (hunk 4 above)
Comment (rbtnn) [reply src/testdir/test_increment.vim]: Is this statement unnecessary?

Comment (k-takata) [reply src/testdir/test_increment.vim]: When I added the num64 feature, the size of Number was depended on the platform. But 8.2.0271 enabled the num64 feature for all platforms. And after that, `v:numbersize` was added in 8.2.0296.
I don't think that the "if" statement is needed. (Or are you concerned about that v:numbersize is more than 64?)

**Comment by brammool** [comment]
> @chrisbra commented on this pull request.
> 
> > @@ -5427,6 +5427,13 @@ A jump table for the options with a short description can be found at |Q_op|.
>  	bin	If included, numbers starting with "0b" or "0B" will be
>  		considered to be binary.  Example: Using CTRL-X on
>  		"0b1000" subtracts one, resulting in "0b0111".
> +	unsigned    If included, numbers is recognized to be unsigned number.
> +		Thus the numbers does not contained negative sign('-').
> 
> I think better would be something like:
> 
> `If included, numbers are recognized as unsigned. Thus a leading negative sign won't be considered as part of the number.`

A better example is something like 9-2020, using CTRL-A on "2020" will
make it "2021" instead of "2019".

Going down from zero to a large number doesn't sound useful.  How about
sticking at zero?

-- 
A bad peace is better than a good war. - Yiddish Proverb

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by rbtnn** [comment]
>A better example is something like 9-2020, using CTRL-A on "2020" will
>make it "2021" instead of "2019".

I'll fix it.

>Going down from zero to a large number doesn't sound useful.  How about
>sticking at zero?

I think so. But present behavior is to go down from -(2^64 - 1) to 2^64 - 1.

**Comment by rbtnn** [comment]
I fixed it.

How about sticking at 2^64 - 1 when increase a large number?

**Comment by brammool** [comment]
> I fixed it.
> 
> How about sticking at 2^64 - 1 when increase a large number?

I don't think it matters.

-- 
I'm in shape.  Round IS a shape.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by rbtnn** [comment]
>> I fixed it.
>>
>> How about sticking at 2^64 - 1 when increase a large number?
>I don't think it matters.

I see. I think  this PR is ready.

**Pull request closed without merging**